homeassistant/components/fujitsu_hvac/__init__.py

```diff
@@ -0,0 +1,63 @@
+"""The Fujitsu HVAC (based on Ayla IOT) integration."""
+
+from __future__ import annotations
+
+from contextlib import suppress
+from dataclasses import dataclass
+
+from ayla_iot_unofficial import AylaAuthError, new_ayla_api
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+
+from .const import API_TIMEOUT, CONF_EUROPE, FGLAIR_APP_ID, FGLAIR_APP_SECRET
+from .coordinator import FujitsuHVACCoordinator
+
+PLATFORMS: list[Platform] = [Platform.CLIMATE]
+
+type FujitsuHVACConfigEntry = ConfigEntry[FujitsuHVACConfigData]
+
+
+@dataclass
+class FujitsuHVACConfigData:
+    """Config data for FujitsuHVAC config entries."""
+
+    coordinator: FujitsuHVACCoordinator
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Fujitsu HVAC (based on Ayla IOT) from a config entry."""
+    api = new_ayla_api(
+        entry.data[CONF_USERNAME],
+        entry.data[CONF_PASSWORD],
+        FGLAIR_APP_ID,
+        FGLAIR_APP_SECRET,
+        europe=entry.data[CONF_EUROPE],
+        timeout=API_TIMEOUT,
+    )
+
+    try:
+        await api.async_sign_in()
+    except TimeoutError as e:
+        raise ConfigEntryNotReady("Timed out while connecting to Ayla IoT API") from e
+    except AylaAuthError as e:
+        raise ConfigEntryAuthFailed("Credentuials expired for Ayla IoT API") from e
```

Comment:
We now have a new function in the coordinator to overwrite `_async_update`. You can move this code into the coordinator now :D

(be sure to sync up your branch with dev)